lib/spack/spack/package.py

```diff
@@ -1420,6 +1524,43 @@ def rpath_args(self):
         return " ".join("-Wl,-rpath,%s" % p for p in self.rpath)
 
 
+class Package(PackageBase):
+    phases = ['install', 'log']
+    # This will be used as a registration decorator in user
+    # packages, if need be
+    PackageBase.sanity_check('install')(PackageBase.sanity_check_prefix)
+
+
+class AutotoolsPackage(PackageBase):
+    phases = ['autoreconf', 'configure', 'build', 'install', 'log']
+
+    def autoreconf(self, spec, prefix):
+        """Not needed usually, configure should be already there"""
+        pass
+
+    @PackageBase.sanity_check('autoreconf')
+    def is_configure_or_die(self):
+        if not os.path.exists('configure'):
+            raise RuntimeError('configure script not found in {0}'.format(os.getcwd()))
+
+    def configure_args(self):
+        return list()
+
+    def configure(self, spec, prefix):
+        options = ['--prefix={0}'.format(prefix)] + self.configure_args()
+        inspect.getmodule(self).configure(*options)
+
+    def build(self, spec, prefix):
+        inspect.getmodule(self).make()
+
+    def install(self, spec, prefix):
+        inspect.getmodule(self).make('install')
+
+    # This will be used as a registration decorator in user
+    # packages, if need be
+    PackageBase.sanity_check('install')(PackageBase.sanity_check_prefix)
```

Comment:
@citibeth A package knows it's phases, so that they could be added to `spack info`. Adding logic to stop at a particular phase of the installation doesn't require every package to follow the same workflow.